### PR TITLE
Added queries and mutations with next-redux-wrapper

### LINKED ISSUE #25: rtk query setup for all the queries and mutations
- setup store for all the queries and mutations that we have

### Queries

- getMembers: `base_url/members`
- getMembersData: `base_url/users/:rds_user_id`
- getContribution: `base_url/contributions/:rds_user_id`
- getUserActiveTask: `base_url/tasks/:rds_user_id/?status=active`

### Mutations

- updateMemberRole:  `base_url}/members/moveToMembers/:rds_user_id
- archiveMember: `base_url/members/archiveMembers/:rds_user_id`
- updateTaskStatus: `base_url/tasks/:task_id

- we need to use next redux wrapper to use SSG SSR in our apps

### PR DESCRIPTION
**Pull request by satyam73**
Closes #25 

I have added Queries and Mutations for the application with the help of next-redux-wrapper.

### CODE REVIEW

File: pages/test.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+interface MyProps {
+  members: Member;
+}
+
+function Test(props:MyProps) {
+  // const { data : members } = useGetMembersQuery(1);
+  const { members } = props;
+  const { data : user } = useGetUserQuery("satyam-bajpai");
```
Comment (vvaibhavdesai) [inline pages/test.tsx:16]: why are we sending your name here ?

File: pages/test.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+interface Member {
+  [key: string]: string;
+}
+
+interface MyProps {
+  members: Member;
+}
+
+function Test(props:MyProps) {
+  // const { data : members } = useGetMembersQuery(1);
```
Comment (vvaibhavdesai) [inline pages/test.tsx:14]: remove this !

**Review by vvaibhavdesai — CHANGES_REQUESTED** [review]

File: pages/test.tsx (hunk 1 above)
Comment (satyam73) [reply pages/test.tsx:16]: Used it for testing purposes😅.
Removed now

File: pages/_app.tsx [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
-import store from "@/src/store";
 import type { AppProps } from "next/app";
+import { wrapper } from "@/src/store";
 
-export default function App({ Component, pageProps }: AppProps) {
-  return  <Provider store = {store}> <Component {...pageProps} /> </Provider> 
-}
\ No newline at end of file
+function App({ Component, ...rest }: AppProps) {
+  const {store, props} = wrapper.useWrappedStore(rest);
```
[the remark this answers is not in the document]
Comment (satyam73) [reply pages/_app.tsx:7]: Comments regarding ?

File: pages/_app.tsx [hunk 4]
```diff
@@ -1,8 +1,15 @@
 import "../styles/globals.css";
```
[the remark this answers is not in the document]
Comment (satyam73) [reply pages/_app.tsx:1]: This was by default present I will lookout for order of imports 👍

File: pages/index.tsx [hunk 5]
```diff
@@ -1,10 +1,27 @@
+import { wrapper } from '@/src/store';
 import Head from 'next/head'
 import Image from 'next/image'
 import styles from '../styles/Home.module.css'
+import Test from './test';
+import serverApi from '../src/services/serverApi';
 
-export default function Home() {
+interface Member {
```
[the remark this answers is not in the document]
Comment (satyam73) [reply pages/index.tsx:8]: Changed this to types

File: pages/index.tsx [hunk 6]
```diff
@@ -1,8 +1,18 @@
...
+import styles from '../styles/Home.module.css';
 
-export default function Home() {
+type PropsType = {
+  pageProps: {
+    members: {
+      [key: string]: string;
+    }
+  }
+}
```
Comment (vinayak-trivedi) [inline pages/index.tsx:13]: just a doubt here, so according to this type, every field in the members object is a value which is a string right?

**Comment by satyam73** [comment]
@vinayak-trivedi take a look into this. I have updated the types for members response.

File: pages/index.tsx [hunk 7]
```diff
@@ -1,8 +1,44 @@
...
-import Image from 'next/image'
-import styles from '../styles/Home.module.css'
+import Head from "next/head";
+import Image from "next/image";
+import { wrapper } from "@/src/store";
+import serverApi from "../src/services/serverApi";
+import styles from "../styles/Home.module.css";
 
-export default function Home() {
+type PropsType = {
```
Comment (vvaibhavdesai) [inline pages/index.tsx:7]: please break them into smaller types

File: pages/index.tsx [hunk 8]
```diff
@@ -17,7 +53,7 @@ export default function Home() {
         </h1>
 
         <p className={styles.description}>
-          Get started by editing{' '}
+          Get started by editing{" "}
```
Comment (vvaibhavdesai) [inline pages/index.tsx:56]: ????

File: pages/index.tsx [hunk 9]
```diff
@@ -60,12 +96,26 @@ export default function Home() {
           target="_blank"
           rel="noopener noreferrer"
         >
-          Powered by{' '}
+          Powered by{" "}
```
Comment (vvaibhavdesai) [inline pages/index.tsx:99]: ?????

File: pages/index.tsx (hunk 7 above)
Comment (satyam73) [reply pages/index.tsx:47]: Got it✅

**Review by vinayak-trivedi — APPROVED** [review]
LGTM

**Review by vvaibhavdesai — APPROVED** [review]
Great work @satyam73 🚀

**Comment by satyam73** [comment]
> Great work @satyam73 🚀

Thanks for guiding throughout the steps🙂 @vvaibhavdesai and @vinayak-trivedi also

**Pull request merged**